src/test/java/com/devonfw/sample/archunit/SecurityTest.java

```diff
@@ -31,4 +57,28 @@ public class SecurityTest {
                     .orShould().beAnnotatedWith(DenyAll.class)
                     .because("All Use-Case implementation methods must be annotated with a security " +
                             "constraint from javax.annotation.security");
+
+    /**
+     * Checks if these methods are being used
+     * Query createQuery(String qlString)
+     * <T> TypedQuery<T> createQuery(String qlString, Class<T> resultClass)
+     * Query createNativeQuery(String sqlString)
+     * Query createNativeQuery(String sqlString, Class resultClass)
+     * Query createNativeQuery(String sqlString, String resultSetMapping)
+     */
+    @ArchTest
+    private static final ArchRule shouldnTUseCreateQuery = noClasses().should().callMethodWhere(new DescribedPredicate<JavaMethodCall>("test whether CreateQuery or CreateNativQuery is used") {
+            @Override
+            public boolean test(JavaMethodCall javaMethod) {
+                        if(javaMethod.getName().equals("createQuery")){
+                                System.out.println(javaMethod.getTarget().getParameterTypes());
```

Comment:
`System.out` may only be used for debugging but shall never be committed ;)

Reply:
sorry, forgot to remove those :)